app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -0,0 +1,177 @@
+import { useState } from 'react'
+import {
+  Alert,
+  CalendarMonth,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  Popover
+} from '@patternfly/react-core'
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon'
+
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import type { FunctionComponent } from 'react'
+
+import './ExpirationDatePicker.scss'
+
+interface ExpirationItem {
+  id: string;
+  label: string;
+  period: number; // In days
+}
+
+const collection: ExpirationItem[] = [
+  { id: '7', label: '7 days', period: 7 },
+  { id: '30', label: '30 days', period: 30 },
+  { id: '60', label: '60 days', period: 60 },
+  { id: '90', label: '90 days', period: 90 },
+  { id: 'custom', label: 'Custom...', period: 0 },
+  { id: 'no-exp', label: 'No expiration', period: 0 }
+]
+
+const today: Date = new Date()
+const tomorrow: Date = new Date(today)
+tomorrow.setDate(today.getDate() + 1)
+const dayMs = 60 * 60 * 24 * 1000
+
+const computeDropdownDate = (dropdownSelectedItem: ExpirationItem) => {
+  if (dropdownSelectedItem.period === 0) return null
+
+  return new Date(today.getTime() + dropdownSelectedItem.period * dayMs)
+}
+
+const computeSelectedDate = (dropdownDate: Date | null, dropdownSelectedItem: ExpirationItem, calendarPickedDate: Date) => {
+  if (dropdownDate) return dropdownDate
+
+  return dropdownSelectedItem.id === 'custom' ? calendarPickedDate : null
+}
+
+const computeFormattedDateValue = (selectedDate: Date | null) => {
+  if (!selectedDate) return ''
+
+  const formatter = Intl.DateTimeFormat('en-US', {
+    month: 'long', day: 'numeric', year: 'numeric', hour: 'numeric', minute: 'numeric', hour12: false
+  })
+
+  return formatter.format(selectedDate)
+}
+
+const computeFieldHint = (formattedDateValue: string) => {
+  if (!formattedDateValue) return
+
+  return `The token will expire on ${formattedDateValue}`
+}
+
+const computeTzMismatch = (tzOffset: number) => {
+  // Timezone offset in the same format as ActiveSupport
+  const jsTzOffset = new Date().getTimezoneOffset() * -60
```

Comment:
Wow, that's confusing :sweat_smile: Not only they are not in the same granularity (seconds vs minutes), but also different signs (negative vs positive) :zany_face: